ui/src/errors.ts

```diff
@@ -6,21 +6,37 @@
 import { localize } from "./localize";
 
 // tslint:disable: max-classes-per-file
+export class AzureToolsError extends Error {
```

Comment:
This should be an interface

Reply:
Preferably with a more generic name that describes what it's adding.

Reply:
Renamed the class.
I kept it as class as it would allow throwing just the base error and doesn't require another derivedclass.

Reply:
What I don't like is that you're exposing this functionality via a requirement to derive from a certain class (I really don't even want users to be able to see another class that suggests this).  Instead, it should be an interface (technically not required to have an interface at all but it's usually helpful).  Users can then simply add the actionContext property to any error, even existing ones that have already been thrown.

Reply:
Not Applicable with the new design